hash_practice/exercises.py

```diff
@@ -2,18 +2,65 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    if len(strings) == 0:
+        return []
+
+    hash = {}
+    for string in strings:
+        #sort out the word alphabetically since we are working with anagrams and make that the key
+        #sorted returns a list and we need to change it back to a string by using join
+        key = ''.join(sorted(string))
+        if key not in hash:
+            #bracket notation to add it in as a list
+            #we're adding in the actual, not sorted, version of the string
+            hash[key] = [string]
+        else:
+            #since the list is already there, append to it
+            hash[key].append(string)
+    print(hash.values())
+    return hash.values()

```

Comment:
This is a correct implementation of the algorithm, but note that `hash.values()` here doesn't return a list, which is why one of the tests is failing. It'd be easy to fix with by making a call to `list`, so `list(hash.values())`.

Not so major that I'm going to dock any points, though.